client/state/purchases/selectors/site-has-search-product-purchase.js

```diff
@@ -8,7 +8,7 @@ import 'calypso/state/purchases/init';
  * Whether a site has an active Jetpack Search purchase.
  *
  * @param   {object} state       global state
- * @param   {number} siteId      the site id
+ * @param   {number|null} siteId      the site id
```

Comment:
why this tho? 🙋🏻‍♂️

Reply:
This is because `useSelector( getSelectedSiteId )` can return null until the state is properly initialized. Also, `getSitePurchases` — the selector `siteHasSearchProductPurchase` primarily relies on — also expects an input type of `number|null`. (In fact, it expects `number|null|undefined`, but we don't expect to pass an undefined into the function in our use-case.)